### PR TITLE
Operation delete for crud

### PR DESCRIPTION
**Pull request by GrekNMD**
Added Delete operation for CRUD and two test to ProductTests

### CODE REVIEW

File: AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs [hunk 1]
```diff
@@ -66,6 +66,69 @@ public async Task POST_AddsItemCorrect(string url)
             Assert.Equal(product.id, addedProduct.id);
         }
 
+        [Theory]
+        [InlineData("/api/product")]
+        public async Task Delete_ItemWithExistingId(string url)
```
Comment (shoroxis) [inline AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs]: Not urgent, but test name structure as following:
<actionToTest>_<condition>_<result>
So in this case possible name is Delete_ItemWithExistingId_ReturnsSuccess

File: AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs [hunk 2]
```diff
@@ -66,6 +66,69 @@ public async Task POST_AddsItemCorrect(string url)
...
+
+            //Try get stored product after deleting 
+            response = await Client.GetAsync(urlWithId);
+            Assert.NotEqual(System.Net.HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+
+        [Theory]
+        [InlineData("/api/product")]
+        public async Task Delete_ItemWithNotExistingId(string url)
```
Comment (shoroxis) [inline AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs]: ...<ReturnsNoContent>

File: AnyServe/AnyServe/Controllers/BaseController.cs [hunk 3]
```diff
@@ -33,7 +33,11 @@ public IEnumerable<T> Get()
         [HttpGet("{id}")]
         public IActionResult Get(Guid id)
         {
-            return Ok(_storage.GetById(id));
+            var s = _storage.GetById(id);
```
Comment (shoroxis) [inline AnyServe/AnyServe/Controllers/BaseController.cs]: Don't use "magic" names like s, better use something like "result"

File: AnyServe/AnyServe/Controllers/BaseController.cs [hunk 4]
```diff
@@ -33,7 +33,11 @@ public IEnumerable<T> Get()
...
+            if(s != null)
+                return Ok(_storage.GetById(id));
```
Comment (shoroxis) [inline AnyServe/AnyServe/Controllers/BaseController.cs]: You execute "GetById" method twice, use previous result instead

File: AnyServe/AnyServe/Storage/Storage.cs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Security.Cryptography.X509Certificates;
```
Comment (shoroxis) [inline AnyServe/AnyServe/Storage/Storage.cs]: What is that? Do we need it?

File: AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs [hunk 6]
```diff
@@ -66,6 +66,69 @@ public async Task POST_AddsItemCorrect(string url)
...
+            Assert.Equal(System.Net.HttpStatusCode.NoContent, response.StatusCode);
+        }
+
+
+        [Theory]
+        [InlineData("/api/product")]
+        public async Task Delete_ItemWithNotExistingId(string url)
+        {
+            // Arrange (test preparation)
+            var product = new Product()
```
Comment (shoroxis) [inline AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs:114]: Actually you don't need any arrange here, just try get non existing item

**Review by shoroxis — CHANGES_REQUESTED** [review]

File: AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs (hunk 6 above)
Comment (GrekNMD) [reply AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs:114]: Need product Id to check if he exist in Repository.

Comment (shoroxis) [reply AnyServe/AnyServe.ITests/ProductModelTests/ProductControllerTests.cs:114]: You can create only Guid for id and not entire person

**Review by shoroxis — APPROVED** [review]

**Comment by GrekNMD** [comment]
Code review passed

**Pull request merged**